### ISSUE TITLE
Bug: NavSatFix.idl: com.eprosima.idl.parser.typecode.AliasTypeCode has no such attribute: cTypeDimensions

### ISSUE THREAD

**Issue by Ryanf55** [eProsima/Micro-XRCE-DDS-Gen#57]
# Summary

This bug reports a illegal identifier when trying to generate NavSatFix.

# Versioning

OS: Ubuntu 22.04
Hash: 0eed89a87427d3faca9a05e61bf31509529d627d
Message: ROS2 `sensor_msgs/msg/NavSatFix`

# Steps to reproduce

1. Take the ROS2 IDL's that make up NavSatFix, dump them in a single folder, and edit the includes to not have folders
2. Run the generator on messages sequentially to build up PoseStamped.
    * ` microxrceddsgen NavSatStatus.idl`
    * ` microxrceddsgen NavSatFix.idl`
4. Observe this error:
```
$ microxrceddsgenNavSatFix.idl
openjdk version "11.0.18" 2023-01-17
OpenJDK Runtime Environment (build 11.0.18+10-post-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 11.0.18+10-post-Ubuntu-0ubuntu122.04, mixed mode, sharing)
Loading templates...
Processing the file libraries/AP_XRCE_Client/Idl/NavSatFix.idl...
WARNING (File NavSatFix, Line 41): Annotation unit not supported. Ignoring...
WARNING (File NavSatFix, Line 46): Annotation unit not supported. Ignoring...
WARNING (File NavSatFix, Line 52): Annotation unit not supported. Ignoring...
WARNING (File NavSatFix, Line 61): Annotation unit not supported. Ignoring...
ERROR: Class com.eprosima.idl.parser.typecode.AliasTypeCode has no such attribute: cTypeDimensions in template context [struct_type anonymous member_declaration elseif(member.typecode.isType_f)_subtemplate]
java.lang.NoSuchFieldException: cTypeDimensions
        at java.base/java.lang.Class.getField(Class.java:1999)
        at org.antlr.stringtemplate.language.ASTExpr.rawGetObjectProperty(ASTExpr.java:524)
        at org.antlr.stringtemplate.language.ASTExpr.getObjectProperty(ASTExpr.java:416)
        at org.antlr.stringtemplate.language.ActionEvaluator.attribute(ActionEvaluator.java:351)
        at org.antlr.stringtemplate.language.ActionEvaluator.expr(ActionEvaluator.java:136)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:84)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ConditionalExpr.writeSubTemplate(ConditionalExpr.java:152)
        at org.antlr.stringtemplate.language.ConditionalExpr.write(ConditionalExpr.java:113)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:722)
        at org.antlr.stringtemplate.language.ASTExpr.writeAttribute(ASTExpr.java:659)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:86)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:722)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:742)
        at org.antlr.stringtemplate.language.ASTExpr.writeAttribute(ASTExpr.java:659)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:86)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.StringTemplate.toString(StringTemplate.java:1635)
        at org.antlr.stringtemplate.StringTemplate.toString(StringTemplate.java:1626)
        at com.eprosima.idl.generator.manager.TemplateGroup.setAttribute(TemplateGroup.java:90)
        at com.eprosima.idl.parser.grammar.IDLParser.definition_list(IDLParser.java:833)
        at com.eprosima.idl.parser.grammar.IDLParser.module(IDLParser.java:748)
        at com.eprosima.idl.parser.grammar.IDLParser.definition(IDLParser.java:517)
        at com.eprosima.idl.parser.grammar.IDLParser.definition_list(IDLParser.java:817)
        at com.eprosima.idl.parser.grammar.IDLParser.module(IDLParser.java:748)
        at com.eprosima.idl.parser.grammar.IDLParser.definition(IDLParser.java:517)
        at com.eprosima.idl.parser.grammar.IDLParser.specification(IDLParser.java:334)
        at com.eprosima.uxr.microxrceddsgen.parseIDL(microxrceddsgen.java:327)
        at com.eprosima.uxr.microxrceddsgen.process(microxrceddsgen.java:264)
        at com.eprosima.uxr.microxrceddsgen.execute(microxrceddsgen.java:179)
        at com.eprosima.uxr.microxrceddsgen.main(microxrceddsgen.java:518)
ERROR: Can't get property cTypename using method get/isCTypename from com.eprosima.idl.parser.typecode.PrimitiveTypeCode instance
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.antlr.stringtemplate.language.ASTExpr.invokeMethod(ASTExpr.java:563)
        at org.antlr.stringtemplate.language.ASTExpr.rawGetObjectProperty(ASTExpr.java:514)
        at org.antlr.stringtemplate.language.ASTExpr.getObjectProperty(ASTExpr.java:416)
        at org.antlr.stringtemplate.language.ActionEvaluator.attribute(ActionEvaluator.java:351)
        at org.antlr.stringtemplate.language.ActionEvaluator.expr(ActionEvaluator.java:136)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:84)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ConditionalExpr.write(ConditionalExpr.java:126)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:722)
        at org.antlr.stringtemplate.language.ASTExpr.writeAttribute(ASTExpr.java:659)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:86)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:722)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:742)
        at org.antlr.stringtemplate.language.ASTExpr.writeAttribute(ASTExpr.java:659)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:86)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.StringTemplate.toString(StringTemplate.java:1635)
        at org.antlr.stringtemplate.StringTemplate.toString(StringTemplate.java:1626)
        at com.eprosima.idl.generator.manager.TemplateGroup.setAttribute(TemplateGroup.java:90)
        at com.eprosima.idl.parser.grammar.IDLParser.definition_list(IDLParser.java:833)
        at com.eprosima.idl.parser.grammar.IDLParser.module(IDLParser.java:748)
        at com.eprosima.idl.parser.grammar.IDLParser.definition(IDLParser.java:517)
        at com.eprosima.idl.parser.grammar.IDLParser.definition_list(IDLParser.java:817)
        at com.eprosima.idl.parser.grammar.IDLParser.module(IDLParser.java:748)
        at com.eprosima.idl.parser.grammar.IDLParser.definition(IDLParser.java:517)
        at com.eprosima.idl.parser.grammar.IDLParser.specification(IDLParser.java:334)
        at com.eprosima.uxr.microxrceddsgen.parseIDL(microxrceddsgen.java:327)
        at com.eprosima.uxr.microxrceddsgen.process(microxrceddsgen.java:264)
        at com.eprosima.uxr.microxrceddsgen.execute(microxrceddsgen.java:179)
        at com.eprosima.uxr.microxrceddsgen.main(microxrceddsgen.java:518)
Caused by: java.lang.IllegalArgumentException: Can't find template type_1e.st; group hierarchy is [Types]
        at org.antlr.stringtemplate.StringTemplateGroup.lookupTemplate(StringTemplateGroup.java:507)
        at org.antlr.stringtemplate.StringTemplateGroup.getInstanceOf(StringTemplateGroup.java:392)
        at org.antlr.stringtemplate.StringTemplateGroup.getInstanceOf(StringTemplateGroup.java:404)
        at com.eprosima.idl.parser.typecode.TypeCode.getCTypenameFromStringTemplate(TypeCode.java:71)
        at com.eprosima.idl.parser.typecode.PrimitiveTypeCode.getCTypename(PrimitiveTypeCode.java:36)
        ... 39 more
ERROR: Can't get property cTypename using method get/isCTypename from com.eprosima.idl.parser.typecode.PrimitiveTypeCode instance
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.antlr.stringtemplate.language.ASTExpr.invokeMethod(ASTExpr.java:563)
        at org.antlr.stringtemplate.language.ASTExpr.rawGetObjectProperty(ASTExpr.java:514)
        at org.antlr.stringtemplate.language.ASTExpr.getObjectProperty(ASTExpr.java:416)
        at org.antlr.stringtemplate.language.ActionEvaluator.attribute(ActionEvaluator.java:351)
        at org.antlr.stringtemplate.language.ActionEvaluator.expr(ActionEvaluator.java:136)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:84)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ConditionalExpr.writeSubTemplate(ConditionalExpr.java:152)
        at org.antlr.stringtemplate.language.ConditionalExpr.write(ConditionalExpr.java:113)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:722)
        at org.antlr.stringtemplate.language.ASTExpr.writeAttribute(ASTExpr.java:659)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:86)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:722)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:742)
        at org.antlr.stringtemplate.language.ASTExpr.writeAttribute(ASTExpr.java:659)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:86)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.StringTemplate.toString(StringTemplate.java:1635)
        at org.antlr.stringtemplate.StringTemplate.toString(StringTemplate.java:1626)
        at com.eprosima.idl.generator.manager.TemplateGroup.setAttribute(TemplateGroup.java:90)
        at com.eprosima.idl.parser.grammar.IDLParser.definition_list(IDLParser.java:833)
        at com.eprosima.idl.parser.grammar.IDLParser.module(IDLParser.java:748)
        at com.eprosima.idl.parser.grammar.IDLParser.definition(IDLParser.java:517)
        at com.eprosima.idl.parser.grammar.IDLParser.definition_list(IDLParser.java:817)
        at com.eprosima.idl.parser.grammar.IDLParser.module(IDLParser.java:748)
        at com.eprosima.idl.parser.grammar.IDLParser.definition(IDLParser.java:517)
        at com.eprosima.idl.parser.grammar.IDLParser.specification(IDLParser.java:334)
        at com.eprosima.uxr.microxrceddsgen.parseIDL(microxrceddsgen.java:327)
        at com.eprosima.uxr.microxrceddsgen.process(microxrceddsgen.java:264)
        at com.eprosima.uxr.microxrceddsgen.execute(microxrceddsgen.java:179)
        at com.eprosima.uxr.microxrceddsgen.main(microxrceddsgen.java:518)
Caused by: java.lang.NullPointerException
        at com.eprosima.idl.parser.typecode.PrimitiveTypeCode.getCTypename(PrimitiveTypeCode.java:36)
        ... 40 more
ERROR: Can't get property cTypename using method get/isCTypename from com.eprosima.idl.parser.typecode.PrimitiveTypeCode instance
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.antlr.stringtemplate.language.ASTExpr.invokeMethod(ASTExpr.java:563)
        at org.antlr.stringtemplate.language.ASTExpr.rawGetObjectProperty(ASTExpr.java:514)
        at org.antlr.stringtemplate.language.ASTExpr.getObjectProperty(ASTExpr.java:416)
        at org.antlr.stringtemplate.language.ActionEvaluator.attribute(ActionEvaluator.java:351)
        at org.antlr.stringtemplate.language.ActionEvaluator.expr(ActionEvaluator.java:136)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:84)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ConditionalExpr.writeSubTemplate(ConditionalExpr.java:152)
        at org.antlr.stringtemplate.language.ConditionalExpr.write(ConditionalExpr.java:113)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:722)
        at org.antlr.stringtemplate.language.ASTExpr.writeAttribute(ASTExpr.java:659)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:86)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:722)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:742)
        at org.antlr.stringtemplate.language.ASTExpr.writeAttribute(ASTExpr.java:659)
        at org.antlr.stringtemplate.language.ActionEvaluator.action(ActionEvaluator.java:86)
        at org.antlr.stringtemplate.language.ASTExpr.write(ASTExpr.java:148)
        at org.antlr.stringtemplate.StringTemplate.write(StringTemplate.java:700)
        at org.antlr.stringtemplate.StringTemplate.toString(StringTemplate.java:1635)
        at org.antlr.stringtemplate.StringTemplate.toString(StringTemplate.java:1626)
        at com.eprosima.idl.generator.manager.TemplateGroup.setAttribute(TemplateGroup.java:90)
        at com.eprosima.idl.parser.grammar.IDLParser.definition_list(IDLParser.java:833)
        at com.eprosima.idl.parser.grammar.IDLParser.module(IDLParser.java:748)
        at com.eprosima.idl.parser.grammar.IDLParser.definition(IDLParser.java:517)
        at com.eprosima.idl.parser.grammar.IDLParser.definition_list(IDLParser.java:817)
        at com.eprosima.idl.parser.grammar.IDLParser.module(IDLParser.java:748)
        at com.eprosima.idl.parser.grammar.IDLParser.definition(IDLParser.java:517)
        at com.eprosima.idl.parser.grammar.IDLParser.specification(IDLParser.java:334)
        at com.eprosima.uxr.microxrceddsgen.parseIDL(microxrceddsgen.java:327)
        at com.eprosima.uxr.microxrceddsgen.process(microxrceddsgen.java:264)
        at com.eprosima.uxr.microxrceddsgen.execute(microxrceddsgen.java:179)
        at com.eprosima.uxr.microxrceddsgen.main(microxrceddsgen.java:518)
Caused by: java.lang.NullPointerException
        at com.eprosima.idl.parser.typecode.PrimitiveTypeCode.getCTypename(PrimitiveTypeCode.java:36)
        ... 40 more
Generating Type definition files...
```

# Message Definition

## ROS2 .msg
```
$ ros2 interface show sensor_msgs/msg/NavSatFix
# Navigation Satellite fix for any Global Navigation Satellite System
#
# Specified using the WGS 84 reference ellipsoid

# header.stamp specifies the ROS time for this measurement (the
#        corresponding satellite time may be reported using the
#        sensor_msgs/TimeReference message).
#
# header.frame_id is the frame of reference reported by the satellite
#        receiver, usually the location of the antenna.  This is a
#        Euclidean frame relative to the vehicle, not a reference
#        ellipsoid.
std_msgs/Header header
        builtin_interfaces/Time stamp
                int32 sec
                uint32 nanosec
        string frame_id

# Satellite fix status information.
NavSatStatus status
        #
        int8 STATUS_NO_FIX =  -1        #
        int8 STATUS_FIX =      0        #
        int8 STATUS_SBAS_FIX = 1        #
        int8 STATUS_GBAS_FIX = 2        #
        int8 status
        uint16 SERVICE_GPS =     1
        uint16 SERVICE_GLONASS = 2
        uint16 SERVICE_COMPASS = 4      #
        uint16 SERVICE_GALILEO = 8
        uint16 service

# Latitude [degrees]. Positive is north of equator; negative is south.
float64 latitude

# Longitude [degrees]. Positive is east of prime meridian; negative is west.
float64 longitude

# Altitude [m]. Positive is above the WGS 84 ellipsoid
# (quiet NaN if no altitude is available).
float64 altitude

# Position covariance [m^2] defined relative to a tangential plane
# through the reported position. The components are East, North, and
# Up (ENU), in row-major order.
#
# Beware: this coordinate system exhibits singularities at the poles.
float64[9] position_covariance

# If the covariance of the fix is known, fill it in completely. If the
# GPS receiver provides the variance of each measurement, put them
# along the diagonal. If only Dilution of Precision is available,
# estimate an approximate covariance from that.

uint8 COVARIANCE_TYPE_UNKNOWN = 0
uint8 COVARIANCE_TYPE_APPROXIMATED = 1
uint8 COVARIANCE_TYPE_DIAGONAL_KNOWN = 2
uint8 COVARIANCE_TYPE_KNOWN = 3

uint8 position_covariance_type
 ```
 ## IDL
 ```
$ cat NavSatFix.idl
// generated from rosidl_adapter/resource/msg.idl.em
// with input from sensor_msgs/msg/NavSatFix.msg
// generated code does not contain a copyright notice

#include "NavSatStatus.idl"
#include "Header.idl"

module sensor_msgs {
  module msg {
    typedef double double__9[9];
    module NavSatFix_Constants {
      @verbatim (language="comment", text=
        "If the covariance of the fix is known, fill it in completely. If the" "\n"        "GPS receiver provides the variance of each measurement, put them" "\n"        "along the diagonal. If only Dilution of Precision is available," "\n"        "estimate an approximate covariance from that.")
      const uint8 COVARIANCE_TYPE_UNKNOWN = 0;
      const uint8 COVARIANCE_TYPE_APPROXIMATED = 1;
      const uint8 COVARIANCE_TYPE_DIAGONAL_KNOWN = 2;
      const uint8 COVARIANCE_TYPE_KNOWN = 3;
    };
    @verbatim (language="comment", text=
      "Navigation Satellite fix for any Global Navigation Satellite System" "\n"
      "" "\n"
      "Specified using the WGS 84 reference ellipsoid")
    struct NavSatFix {
      @verbatim (language="comment", text=
        "header.stamp specifies the ROS time for this measurement (the" "\n"
        "       corresponding satellite time may be reported using the" "\n"
        "       sensor_msgs/TimeReference message)." "\n"
        "" "\n"
        "header.frame_id is the frame of reference reported by the satellite" "\n"
        "       receiver, usually the location of the antenna.  This is a" "\n"
        "       Euclidean frame relative to the vehicle, not a reference" "\n"
        "       ellipsoid.")
      std_msgs::msg::Header header;

      @verbatim (language="comment", text=
        "Satellite fix status information.")
      sensor_msgs::msg::NavSatStatus status;

      @verbatim (language="comment", text=
        "Latitude. Positive is north of equator; negative is south.")
      @unit (value="degrees")
      double latitude;

      @verbatim (language="comment", text=
        "Longitude. Positive is east of prime meridian; negative is west.")
      @unit (value="degrees")
      double longitude;

      @verbatim (language="comment", text=
        "Altitude. Positive is above the WGS 84 ellipsoid" "\n"
        "(quiet NaN if no altitude is available).")
      @unit (value="m")
      double altitude;

      @verbatim (language="comment", text=
        "Position covariance defined relative to a tangential plane" "\n"
        "through the reported position. The components are East, North, and" "\n"
        "Up (ENU), in row-major order." "\n"
        "" "\n"
        "Beware: this coordinate system exhibits singularities at the poles.")
      @unit (value="m^2")
      double__9 position_covariance;

      uint8 position_covariance_type;
    };
  };
};
````

**Comment by Acuadros95** [comment]
The problem is related to this typedef on the generated idl: `typedef double double__9[9];`

This kind of declaration for an array is not currently supported by the generator.

Check out this fix where the `typedef` is ignored and the type is used as is: https://github.com/eProsima/Micro-XRCE-DDS-Gen/pull/58

**Comment by Acuadros95** [comment]
Closed by #58
